src/uwtools/drivers/jedi.py

```diff
@@ -0,0 +1,135 @@
+"""
+A driver for the jedi component.
+"""
+
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from iotaa import asset, dryrun, refs, run, task, tasks
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class JEDI(Driver):
+    """
+    A driver for the JEDI component.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @task
+    def configuration_file(self):
+        """
+        The configuration file.
+        """
+        fn = "jedi.yaml"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        self._create_user_updated_config(
+            config_class=YAMLConfig,
+            config_values=self._driver_config["configuration_file"],
+            path=path,
+        )
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.configuration_file(),
+            self.files_copied(),
+            self.files_linked(),
+            self.runscript(),
+        ]
```

Comment:
I think that, if we add `self.validate_only()` to this list, we'll get validation of the config file automatically when we run the `provisioned_run_directory` task.

Reply:
Question on this, the assert in the test cases call:
```
def test_JEDI_provisioned_run_directory(driverobj):
    with patch.multiple(
        driverobj,
        files_copied=D,
        files_linked=D,
        runscript=D,
        configuration_file=D,
    ) as mocks:
        driverobj.provisioned_run_directory()
    for m in mocks:
        mocks[m].assert_called_once_with()
```

Here now that the `validate_only` task calls `configuration_file` task. We are actually calling `configuration_file` task twice. I was looking into the `unittest.mock` references to `assert_called_...` and there are a couple options [here](https://docs.python.org/3/library/unittest.mock.html). Should we use `assert_has_calls` for the `configuration_file` task and if so I'm wondering what these would look like.